mathesar_ui/src/api/tables/records.ts

```diff
@@ -0,0 +1,46 @@
+export type ResultValue = string | number | boolean | null;
```

Comment:
We can rename this file to `records.d.ts` to keep it consistent with other type definition files.

The directory for this could be something other than `api`, like `endpoint-types`. `api` gives an impression that we'd be making api calls from here. We could also potentially move all our `api` calls here in a later refactoring, I'm open to that option as well. @seancolsen  We can discuss further on this.

We can go ahead and merge this PR. These changes are minor can be taken up later.